widget.subscribe.php

```diff
@@ -299,6 +354,15 @@ function add_subscriber() {
 
 		}
 
+		// set the source
+		if ( isset( $_POST['source'] ) && !empty( $_POST['source'] ) ) {
```

Comment:
I believe that will throw a warning. Which will not pass Automattic's code review, but I'll check.